Redninja/Components/Conditions/EventCondition.cs

```
@@ -0,0 +1,32 @@
+﻿using System;
+using System.Collections.Generic;
+using Redninja.Components.Conditions.Expressions;
+using Redninja.Events;
+
+namespace Redninja.Components.Conditions
+{
+	internal class EventCondition : ConditionBase
+	{
+		public EventCondition(IInitialExpression left, IInitialExpression right, IConditionalOperator op, IOperatorCountRequirement req)
+		{
+			this.Left = left;
+			this.Right = right;
+			this.Op = op;
+			this.OpRequirement = req;
+		}
+
+		public bool IsTargetConditionMet(IUnitModel self, IBattleEvent battleEvent, IBattleModel battleModel)
+		{
+			IUnitModel target = battleEvent is ITargetedEvent ? ((ITargetedEvent)battleEvent).Target : null;
+			ExpressionResolver resolver = new ExpressionResolver(self, target, battleModel, battleEvent);
```

Is `ExpressionResolver` null-safe for the `target` parameter?

---

This is explained in my [Challenges](https://github.com/riftgarret/ludum/wiki/condition-challenges) section. I think maybe we could pick a solution whether we ignore it, or throw an exception or just secretly print it to log hoping that the case where this was used was intentional.

---

It might be fine to just leave it, maybe throw an exception with an explanation inside whichever expression needs it. That way in case we define some invalid expression we at least can see a helpful exception instead of a random `NullReferenceException`.